src/cms-content/vaccines/index.ts

```diff
@@ -0,0 +1,66 @@
+import { useState, useEffect } from 'react';
+import { getStateCode, Region, RegionType, State } from 'common/regions';
+
+export interface RegionVaccinationInfo {
+  hidden: boolean;
+  fips: string;
+  locationName: string;
+  stateCode: string;
+  eligibilityInfoUrl: string;
+  vaccinationSignupUrl?: string;
+}
+
+export interface RegionTypeVaccinationInfo {
+  regions: RegionVaccinationInfo[];
+}
+
+const fetchRegionTypeVaccineData = (regionType: RegionType) => {
+  if (regionType !== RegionType.STATE) {
+    return;
+  }
+  return fetch(`/cms-content/vaccines/${regionType}.json`).then(
+    res => res.json() as Promise<RegionTypeVaccinationInfo>,
+  );
+};
+
+export const useRegionTypeVaccineData = (regionType: RegionType) => {
+  const [data, setData] = useState<RegionVaccinationInfo[]>();
+  useEffect(() => {
+    fetchRegionTypeVaccineData(regionType)?.then(
+      (value: RegionTypeVaccinationInfo) => {
+        setData(value.regions);
+      },
+    );
+  }, [regionType]);
+  return data;
+};
+
+export const useRegionsVaccineData = (
+  regionType: RegionType,
+  regions: Region[],
+) => {
+  const [data, setData] = useState<RegionVaccinationInfo[]>();
+  const regionTypeData = useRegionTypeVaccineData(regionType);
+  useEffect(() => {
+    const matching =
+      regionTypeData?.filter(item =>
+        regions.map(region => region.fipsCode).includes(item.fips),
+      ) || [];
+    setData(matching);
+  }, [regionTypeData, regions]);
```

Comment:
It might be easier to create a map with the vaccination information by flips to avoid iterating through the `regionTypeData` array for each lookup

Reply:
I think i might end up redoing how this is called, so going to punt this for now!